tensorflow/tools/compatibility/ipynb.py

```diff
@@ -62,8 +62,37 @@ def process_file(in_filename, out_filename, upgrader):
   return files_processed, report_text, errors
 
 
+def skip_magic(code_line, magic_list):
+  """
+  Checks if the cell has magic, that is not python-based.
+
+   >>> skip_magic('!ls -laF', ['%', '!', '?'])
+  True
+  """
+
+  for magic in magic_list:
+    if code_line.startswith(magic):
+      return True
+
+  return False
+
+
+def check_line_split(code_line):
+  r"""
+  Checks if a line was splitted with `\`.
+
+  >>> skip_magic("!gcloud ml-engine models create ${MODEL} \\\n")
+  True
+  """
+
+  if code_line.endswith('\\\n'):
```

Comment:
Will this catch lines with trailing whitespace (which I think is allowed after `\`)?